contents/handbook/engineering/clickhouse/clusters.mdx

```diff
@@ -0,0 +1,220 @@
+### Instance types
+
+The original nodes of the cluster are using `i3en.12xlarge` instances. We are currently in the process of migrating to `im4gn.16xlarge` instances.
+
+#### Online cluster
+
+- 2 shards
+- 2 replicas
+- `i3en.12xlarge` instances
+- 48 vCPUs
+- Intel Xeon CPU
+- 384 GiB RAM
+- Max Network 50 Gbps
+- Max EBS throughput 1187.5 MB/s
+- 4 x 7.5 TB NVMe SSD
+    - RAID 10 far layout
+    - md0 volume
+- 3 x 10TB GP3 EBS volumes
+    - JBOD configuration
+- 1 x 16TB GP3 EBS volume
+    - JBOD configuration
+
+
+#### Offline cluster
+
+- 2 shards
+- 1 replica
+- `im4gn.16xlarge` instances
+- 64 vCPUs
+- Graviton2 CPU
+- 256 GiB RAM
+- Max Network 100 Gbps
+- Max EBS Throughput 5000 MB/s
+- 4 x 7.5 TB NVMe SSD
+    - RAID 10 far layout
+    - md0 volume
+- 3 x 16TB GP3 EBS volumes
+    - JBOD configuration
+- 1 x 1TB GP3 EBS volume (Default)
+
+Old nodes are using `r6i.16xlarge` instances. These are being retired due to IO throughput constraints.
+
+### Tiered storage
+
+One of the nice features of our data is that recent data is the hottest both for reads and writes. This is a perfect fit for tiered storage.
+We can basically read and write to local ephemeral NVMe (with solid backup strategies) and then move the data to cheaper EBS volumes as it ages out.
+We currently do this using tiered storage configured simply by setting the `storage` configs in ClickHouse, but we eventually will want to move to setting TTLs on tables and having ClickHouse manage the tiering for us consistently.
+
+https://altinity.com/blog/2019-11-29-amplifying-clickhouse-capacity-with-multi-volume-storage-part-2
+
+### Monitoring
+
+https://grafana.prod-us.posthog.dev/d/vm-clickhouse-cluster-overview/clickhouse-cluster-overview?from=now-3h&to=now&timezone=utc&refresh=30s
+
+
+## Prod-EU
+
+Prod EU is our main production cluster for the EU.
+
+It is made up of the following setup:
+
+- 8 shards
+- 2 replicas
+- `m6g.8xlarge`
+- 32 vCPUs
+- Graviton2 CPU
+- 128 GiB RAM
+- Max Network 12 Gbps
+- Max EBS Throughput 1187.5 MB/s
+- Single 10TB GP3 EBS volume 
+
+
+We hit a problem with having smaller shards on EU that had a significant performance impact. We were running out of memory for larger queries which was also impacting our page cache hit rate.
+This was mainly due to limiting query size restrictions to protect other users of the cluster.
+We had two solutions for this. Increase the size of the nodes...meaning double the size for each instance x 16, very expensive. Or, we could setup a coordinator node.
+The coordinator node is a topology that allows us to effectively split the storage and compute tiers of the cluster into two pools of resources.
+We treat the current cluster of small nodes with many shards as the storage tier, they effectively are the mappers of the cluster and quickly fetch the data that we want for queries.
+We then send that relatively small data back to the coordinator and do the heavy lifting there which includes joins and aggregates.
```

Comment:
In the US cluster 1 of the replicas is considered 'Offline' for background tasks. Do we have something similar here to avoid impacting the PostHog app?

Reply:
No, we haven't hit the same issues on EU yet. We will probably have a separate set of Coordinators for these kinds of tasks is my inuition though. Totally open to thoughts on it though!

Reply:
Ok! I was thinking about that, the topology really allows it. Thanks!